From 4347567c529a11af4fad0d41cd67ee1c60ffcade Mon Sep 17 00:00:00 2001
From: Alexander Linder <alexander.linder@mondcarion.group>
Date: Tue, 26 Mar 2024 14:26:16 +0100
Subject: [PATCH 1/2] Fix permissions after copying templates

---
 opp_env/opp_env.py | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/opp_env/opp_env.py b/opp_env/opp_env.py
index 47e5d66..c1d7277 100755
--- a/opp_env/opp_env.py
+++ b/opp_env/opp_env.py
@@ -804,6 +804,7 @@ class Workspace:
         if re.search("\\s", dir):
             raise Exception(f"Whitespace characters are not allowed in the name and path of the workspace directory")
         shutil.copytree(os.path.join(os.path.dirname(__file__), "templates", "workspace"), opp_env_dir)
+        reset_permissions(opp_env_dir)
         if nixless:
             # write an empty file called .nixless to indicate that this is a nixless workspace
             open(os.path.join(opp_env_dir, ".nixless"), "w").close()
@@ -1530,6 +1531,18 @@ def is_subdirectory(child_dir, parent_dir):
     # Check if a directory is a subdirectory of another directory.
     return os.path.commonpath([child_dir, parent_dir]) == parent_dir
 
+def reset_permissions(path):
+    for root, dirs, files in os.walk(path):
+        # Reset permissions for directories
+        for directory in dirs:
+            dir_path = os.path.join(root, directory)
+            os.chmod(dir_path, 0o755)
+
+        # Reset permissions for files
+        for file in files:
+            file_path = os.path.join(root, file)
+            os.chmod(file_path, 0o644)
+
 def check_multiple_versions(project_descriptions):
     name_versions_dict = {}
     for project_description in project_descriptions:
-- 
2.43.2

