From 2d32238249c97a12689e6ec2786ce9d04b739655 Mon Sep 17 00:00:00 2001
From: Alexander Linder <alexander.linder@mondcarion.group>
Date: Tue, 26 Mar 2024 14:27:03 +0100
Subject: [PATCH 2/2] Remove upgrade subcommand

---
 opp_env/opp_env.py | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/opp_env/opp_env.py b/opp_env/opp_env.py
index c1d7277..fd2f20f 100755
--- a/opp_env/opp_env.py
+++ b/opp_env/opp_env.py
@@ -312,11 +312,6 @@ def create_arg_parser():
         "local"
     ])
 
-    subparser = subparsers.add_parser("upgrade", help="Upgrades opp_env to the latest version", description="Upgrades opp_env to the latest version")
-    subparser.add_argument("-n", "--dry-run", default=False, action='store_true', help="Test if an upgrade is available and simulate the upgrade without actually carrying out the installation")
-    subparser.add_argument("-c", "--check", default=False, action='store_true', help="Only test if an upgrade is available")
-    subparser.add_argument("--from-pypi", default=False, action='store_true', help="Use the latest version from PyPI instead of the latest version from the Git repository")
-
     subparser = subparsers.add_parser("maint", help="Maintenance functions", description="Maintenance functions")
     subparser.add_argument("-u", "--update-catalog", metavar="download-items-dir", dest="catalog_dir", help="Update the opp_env installation commands in the model catalog of omnetpp.org. The argument should point to the `download-items/` subdir of a checked-out copy of the https://github.com/omnetpp/omnetpp.org/ repository.")
 
@@ -1643,21 +1638,6 @@ def run_subcommand_main(projects, command=None, workspace_directory=None, reques
     _logger.info(f"Running {'test ' if run_test else 'smoke_test ' if run_smoke_test else ''}command for projects {cyan(str(effective_project_descriptions))} in workspace {cyan(workspace.root_directory)} in {cyan(kind)} mode")
     workspace.nix_develop(effective_project_descriptions, workspace_directory, commands=commands, **dict(kwargs, suppress_stdout=False))
 
-def upgrade_subcommand_main(dry_run=False, from_pypi=False, check=False, **kwargs):
-    latest_version = get_latest_version_from_pypi() if from_pypi else get_latest_version_from_github()
-    version = get_version()
-    upgrade_needed = Version(version) < Version(latest_version)
-    _logger.info(f"An upgrade is available for opp_env" if upgrade_needed else f"opp_env is up-to-date")
-    _logger.info(f"Installed version: {cyan(version)}, latest version: {cyan(latest_version)}")
-    if check:
-        return
-    dry_run_option = "--dry-run" if dry_run else ""
-    module_spec = f"opp_env=={latest_version}" if from_pypi else f"git+https://github.com/omnetpp/opp_env.git@{latest_version}"
-    if upgrade_needed:
-        upgrade_command = f"pip3 install --user --upgrade --break-system-packages {dry_run_option} {module_spec}"
-        _logger.debug(f"Executing command: {upgrade_command}")
-        subprocess.run(["bash", "-c", upgrade_command])
-
 def maint_subcommand_main(catalog_dir, **kwargs):
     update_catalog(catalog_dir)
 
@@ -1718,8 +1698,6 @@ def main():
             shell_subcommand_main(**kwargs)
         elif subcommand == "run":
             run_subcommand_main(**kwargs)
-        elif subcommand == "upgrade":
-            upgrade_subcommand_main(**kwargs)
         elif subcommand == "maint":
             maint_subcommand_main(**kwargs)
         else:
-- 
2.43.2

